scripts/py-style-check.sh

```diff
@@ -3,8 +3,8 @@
 set -euo pipefail
 
 # TODO: stop ignoring this. Maybe autopep8 existing stuff?
-find tools -type f -name "*.py" | xargs pep8 -r --show-source --ignore=E123,E125,E126,E127,E128,E302 || \
-    echo "pep8 run failed, please fix it" >&2
+find tools -type f -name "*.py" | xargs flake8 || echo "flake8 run failed, please fix it" >&2
+flake8 . || echo "flake8 run failed, please fix it" >&2
```

Comment:
The second line makes the first useless.

Could we restrict this to `tools` (and maybe a few other directories)?

Reply:
Yes.  I was experimenting in the vain hope that the CI would actually run.  We are looking for syntax errors.  Probably better that the linter finds them in our examples than our users do.